### PR TITLE
[Merged by Bors] - Add NumUnits and VRFNonce to InitialPost

### PR DESCRIPTION
**Pull request by fasmat**
## Motivation
Extends the initial post for the ATX builder with NumUnits and VRFNonce

## Changes
- add `NumUnits` and `VRFNonce` to InitialPost in local DB

## Test Plan
- n/a

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [x] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

File: sql/migrations/local/0001_initial.sql [hunk 1]
```diff
@@ -18,5 +18,7 @@ CREATE TABLE initial_post
     post_indices  VARCHAR NOT NULL,
     post_pow      UNSIGNED LONG INT NOT NULL,
 
-    commit_atx    CHAR(32)
+    num_units     UNSIGNED INT NOT NULL,
+    commit_atx    CHAR(32) NOT NULL,
+    vrf_nonce     UNSIGNED LONG INT NOT NULL
```
Comment (poszu) [inline sql/migrations/local/0001_initial.sql]: You didn't decide to include the labels per unit here. It makes sense as it is part of the network config (hence same for everybody, at least so far). On the other hand, it is used in the POST verification flow and is not required to be equal to the network config - it must be greater or equal (although it's unclear why anybody would use a bigger value).

We should probably either include labels_per_unit here to be consistent or start wiping out using labels_per_unit from POST validation (use the value from the config instead).

**Review by poszu — APPROVED** [review]

File: sql/migrations/local/0001_initial.sql (hunk 1 above)
Comment (dshulyak) [reply sql/migrations/local/0001_initial.sql]: once local database is released it is not possible to extend initial migration

Comment (poszu) [reply sql/migrations/local/0001_initial.sql]: Labels per unit field is supposed to be removed from ATX: https://github.com/spacemeshos/go-spacemesh/issues/4767. So let's not add it here.

> once local database is released it is not possible to extend initial migration

I think @mafa wants to include these fields in the initial release of the local state.

Comment (dshulyak) [reply sql/migrations/local/0001_initial.sql]: but v1.3 branch is already there with migration for local db. we don't use it yet?

Comment (poszu) [reply sql/migrations/local/0001_initial.sql]: v1.3 is not released yet, or is it?

Comment (dshulyak) [reply sql/migrations/local/0001_initial.sql]: it is not released, it has release candidate and branch already there. i don't understand though, @fasmat plans to cherry pick this change to that branch before it is released? thats something weird

Comment (fasmat) [reply sql/migrations/local/0001_initial.sql]: - I was not planning on including this change in `v1.3` but in the next version.
- I will update this to be its separate database migration and add labels per unit
- Removing labes per unit from this state and from ATXs is a bigger change that should be tackled in a dedicated PR that closes #4767

**Review by poszu — APPROVED** [review]

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**